fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/Dimension.java

```diff
@@ -75,6 +75,8 @@
      * @param name  field name
      *
      * @return DimensionField
+     *
+     * @throws IllegalArgumentException if this dimension does not have a field with the specified name
```

Comment:
This should depend on implementation right? so may be move it to key value store dimension? Or are we trying to force people to follow this contract?

Reply:
The [ApiFilter](https://github.com/yahoo/fili/blob/c930a5858291b0eb3d98bc62009144c359c9ef1b/fili-core/src/main/java/com/yahoo/bard/webservice/web/ApiFilter.java#L130) relies on this behavior to validate properly, so it very much *can't* depend on the implementation.